gamestate/user.go

```diff
@@ -45,20 +50,75 @@ func NewUserGateway(gfClient GFClient, loginSocket LoginSocket, gameSocket GameS
 	}
 }
 
+type GfAccountData struct {
+	Email    string
+	Password string
+	Locale   string
+	Name     string
+}
+
 // GameClientGateway provides methods to retrieve information about
 // the real game client.
 type GameClientGateway interface {
 	Version() string
 	Hash() string
 }
 
-func (ug *UserGateway) AuthenticateGFClient(user actions.User, serverLang string) (string, error) {
-	token, accountID, err := ug.gfClient.Authenticate(user, serverLang)
+func (ug *UserGateway) AuthenticateGFClient(jsonAccountPath string, jsonIdentityPath string) (string, error) {
```

Comment:
It's better to pass entities/models directly (User) instead of passing a config path here since we're working with abstractions at this level of the architecture. Ideally, the config files should be read by the bot that uses noskit as a library and pass down the configuration parameters as needed (e.g. the fields contained in the json files as represented by the GfAccountData you created).

This allows us to test the single components in isolation and prevents dependencies from crossing architectural boundaries.